internal/argocd/health.go

```diff
@@ -0,0 +1,277 @@
+package argocd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"slices"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	argocd "github.com/akuity/kargo/internal/controller/argocd/api/v1alpha1"
+	"github.com/akuity/kargo/internal/git"
+)
+
+// healthErrorConditions are the v1alpha1.ApplicationConditionType conditions
+// that indicate an Argo CD Application is unhealthy.
+var healthErrorConditions = []argocd.ApplicationConditionType{
+	argocd.ApplicationConditionComparisonError,
+	argocd.ApplicationConditionInvalidSpecError,
```

Comment:
I could use someone with Argo expertise to tell me if there are any other errors which would fall into this category.

Reply:
I cannot think of any others. We can start with this and improve it later if any others arise.